python/GangaLHCb/Lib/Applications/GaudiExec.py

```diff
@@ -1,31 +1,42 @@
-from os import rename, path, makedirs, chdir, unlink, listdir
+from os import rename, path, makedirs, chdir, unlink, listdir, chmod
+from os import stat as os_stat
 import tempfile
 import time
 import subprocess
 import shutil
+import tarfile
+import threading
+import stat
+from StringIO import StringIO
 
 from Ganga.Core import ApplicationConfigurationError, ApplicationPrepareError, GangaException
+from Ganga.GPIDev.Adapters.IGangaFile import IGangaFile
 from Ganga.GPIDev.Adapters.IPrepareApp import IPrepareApp
+from Ganga.GPIDev.Base.Filters import allComponentFilters
 from Ganga.GPIDev.Base.Proxy import getName
 from Ganga.GPIDev.Lib.File.File import ShareDir
 from Ganga.GPIDev.Lib.File.LocalFile import LocalFile
 from Ganga.GPIDev.Schema import Schema, Version, SimpleItem, GangaFileItem
 from Ganga.Utility.logging import getLogger
-from Ganga.Utility.files import expandfilename
+from Ganga.Utility.files import expandfilename, fullpath
 
 from GangaDirac.Lib.Files.DiracFile import DiracFile
 from GangaDirac.Lib.Backends.DiracBase import DiracBase
 
-from .GaudiRunAppUtils import getGaudiRunInputData, _exec_cmd
+from .GaudiExecUtils import getGaudiExecInputData, _exec_cmd, getTimestampContent, gaudiPythonWrapper
 
 logger = getLogger()
 
-
-class GaudiRun(IPrepareApp):
+class GaudiExec(IPrepareApp):
     """
+
     Welcome to the new GaudiApp for LHCb apps written/constructed making use of the new CMake framework
 
+    Requirements:
+    ===============
```

Comment:
In [RestructuredText](http://www.sphinx-doc.org/en/stable/rest.html#sections), the header underlines should match the length of the title.